src/neo/SmartContract/Native/NeoToken.cs

```diff
@@ -63,9 +63,14 @@ public override BigInteger TotalSupply(DataCache snapshot)
             return TotalAmount;
         }
 
-        internal override async ContractTask OnBalanceChanging(ApplicationEngine engine, UInt160 account, NeoAccountState state, BigInteger amount)
+        internal override void OnBalanceChanging(ApplicationEngine engine, UInt160 account, NeoAccountState state, BigInteger amount)
         {
-            await DistributeGas(engine, account, state);
+            GasDistribution distribution = DistributeGas(engine, account, state);
+            if (distribution is not null)
+            {
+                var list = engine.CurrentContext.GetState<List<GasDistribution>>();
+                list.Add(distribution);
```

Comment:
We append to the list even if there is something already there for the `account`?

Reply:
Yes, this is the same as the original logic, one record is one record, no summary is made.